uWeather/app/src/main/java/dit257/mandalore/uweather/Probability.kt

```diff
@@ -1,36 +1,36 @@
 package dit257.mandalore.uweather
 
+import java.lang.Math.*
 import java.util.*
+import kotlin.math.abs
 
 class Probability {
 
     //Calculate probability based on difference from median
-    fun calcProbabilityMedian(all_last_temps:IntArray): Double {
-        var total = 0
-        for(t in all_last_temps){
+    fun calcProbabilityMean(allLastTemps:DoubleArray, temp:Double): Double {
+        var total = 0.0
+        for(t in allLastTemps){
             total += t
         }
-        return total / 4.0 % 10
+        val mean = total / allLastTemps.size
+        return 1 - (abs(temp - mean) / 100)
     }
-    //Calculate probability based on providers track record
-    fun calcProbabilityTrackRecord(provider:LinkedList<Int>, actual_temp:LinkedList<Int>):Double{
-        var difference = 0.0
-        var nrOfReadings = 0
-        var max = 0
-        var min = 0
-        for (i in provider){
-            //Get min/max temperature reported
-            if(i<min){min=i}
-            if(actual_temp.get(nrOfReadings-1)<min){min=i}
-            if(i>max){max=i}
-            if(actual_temp.get(nrOfReadings-1)>max){max=i}
-
-            nrOfReadings++
-            difference += i - actual_temp.get(nrOfReadings-1)
+    // Calculate the span of temperatures, with a confidence interval of 90%.
```

Comment:
please make a more clear description that it takes an array of temperatures from different weather services to calculate a probable (is aggregated the right word?)  confidence intervall.

Reply:
Okay, I can update the comments and see if it is more understandable